kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/ModemTaskManager.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.nm;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.kura.nm.signal.handlers.NMModemTaskHandler;
+import org.freedesktop.dbus.connections.impl.DBusConnection;
+import org.freedesktop.dbus.exceptions.DBusException;
+import org.freedesktop.networkmanager.Device;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ModemTaskManager {
+
+    private static final Logger logger = LoggerFactory.getLogger(ModemTaskManager.class);
+
+    private final DBusConnection dbusConnection;
+    private final Map<String, NMModemTaskHandler> modemTaskHandlers = new HashMap<>();
+
+    public ModemTaskManager(DBusConnection dbusConnection) {
+        this.dbusConnection = dbusConnection;
+    }
+
+    protected void modemTaskHandlerEnable(String deviceId, Device device, NetworkProperties properties)
+            throws DBusException {
+        int resetDelayMinutes = properties.get(Integer.class, "net.interface.%s.config.resetTimeout", deviceId);
+        boolean autoconnect = properties.get(Boolean.class, "net.interface.%s.config.persist", deviceId);
+        if (isModemTaskAlreadyActivated(deviceId, resetDelayMinutes, autoconnect)) {
+            return;
+        }
+        modemTaskHandlerDisable(deviceId);
+
+        logger.info("Modem {} activated. Starting monitoring task...", deviceId);
+        ModemTaskScheduler connectionScheduler = new ModemTaskScheduler(deviceId, device, properties);
+        if (autoconnect) {
+            connectionScheduler.scheduleConnection();
+        }
+        if (resetDelayMinutes > 0) {
+            connectionScheduler.scheduleReset();
+        }
+        NMModemTaskHandler modemConnectionHandler = new NMModemTaskHandler(connectionScheduler);
+
+        this.modemTaskHandlers.put(deviceId, modemConnectionHandler);
+
+        if (autoconnect || resetDelayMinutes > 0) {
+            this.dbusConnection.addSigHandler(org.freedesktop.networkmanager.Device.StateChanged.class,
+                    modemConnectionHandler);
+        }
```

Comment:
If you move this inside the `if` block you won't need https://github.com/eclipse-kura/kura/pull/5762/files#r2034793463

```suggestion
        NMModemTaskHandler modemConnectionHandler = new NMModemTaskHandler(connectionScheduler);



        if (autoconnect || resetDelayMinutes > 0) {
            this.modemTaskHandlers.put(deviceId, modemConnectionHandler);
            this.dbusConnection.addSigHandler(org.freedesktop.networkmanager.Device.StateChanged.class,
                    modemConnectionHandler);
        }
```